### PR TITLE
[DEV-2234] add asset instant/card cols

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/shared/models/asset/asset.entity.ts
Comment (davidleomay) [inline src/shared/models/asset/asset.entity.ts:1]: Check the new fields in PaymentInfoService

File: src/shared/services/payment-info.service.ts [hunk 1]
```diff
@@ -21,13 +21,7 @@ export class PaymentInfoService {
...
       if (!dto.currency) throw new NotFoundException('Currency not found');
-      if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.CARD) {
-        if (!dto.currency.cardSellable) throw new BadRequestException('Currency not sellable');
-      } else if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.INSTANT) {
-        if (!dto.currency.instantSellable) throw new BadRequestException('Currency not sellable');
-      } else {
-        if (!dto.currency.sellable) throw new BadRequestException('Currency not sellable');
-      }
+      if (!dto.currency.sellable) throw new BadRequestException('Currency not sellable');
     }
```
Comment (davidleomay) [inline src/shared/services/payment-info.service.ts:25]: This check should only be made if PaymentMethod is Bank (`sellable` = `bankSellable` 😉 )

File: src/shared/services/payment-info.service.ts [hunk 2]
```diff
@@ -36,6 +30,14 @@ export class PaymentInfoService {
     if (jwt && !jwt.blockchains.includes(dto.asset.blockchain))
       throw new BadRequestException('Asset blockchain mismatch');
 
+    if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.CARD) {
+      if (!dto.currency.cardSellable) throw new BadRequestException('Currency not sellable per Card');
+      if (!dto.asset.cardSellable) throw new BadRequestException('Asset not buyable per Card');
+    } else if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.INSTANT) {
+      if (!dto.currency.instantSellable) throw new BadRequestException('Currency not sellable per Instant');
+      if (!dto.asset.instantSellable) throw new BadRequestException('Asset not buyable per Instant');
+    }
```
Comment (davidleomay) [inline src/shared/services/payment-info.service.ts:39]: There might be no currency in the DTO here?

File: src/shared/services/payment-info.service.ts [hunk 3]
```diff
@@ -36,6 +30,14 @@ export class PaymentInfoService {
     if (jwt && !jwt.blockchains.includes(dto.asset.blockchain))
       throw new BadRequestException('Asset blockchain mismatch');
 
+    if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.CARD) {
+      if (!dto.currency.cardSellable) throw new BadRequestException('Currency not sellable per Card');
+      if (!dto.asset.cardSellable) throw new BadRequestException('Asset not buyable per Card');
+    } else if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.INSTANT) {
+      if (!dto.currency.instantSellable) throw new BadRequestException('Currency not sellable per Instant');
+      if (!dto.asset.instantSellable) throw new BadRequestException('Asset not buyable per Instant');
```
Comment (davidleomay) [inline src/shared/services/payment-info.service.ts:38]: `per` sounds a bit german 😁  maybe use `via`?

File: src/shared/services/payment-info.service.ts (hunk 2 above)
Comment (Yannick1712) [reply src/shared/services/payment-info.service.ts:39]: Nope not possible in this case I think. Currency IsNotEmptyObject in dto's with paymentMethod 👍

File: src/shared/services/payment-info.service.ts [hunk 4]
```diff
@@ -21,21 +21,25 @@ export class PaymentInfoService {
...
       throw new BadRequestException('Asset blockchain mismatch');
 
+    if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.CARD) {
+      if (!dto.currency.cardSellable) throw new BadRequestException('Currency not sellable via Card');
+      if (!dto.asset.cardBuyable) throw new BadRequestException('Asset not buyable via Card');
+    } else if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.INSTANT) {
+      if (!dto.currency.instantSellable) throw new BadRequestException('Currency not sellable via Instant');
+      if (!dto.asset.instantBuyable) throw new BadRequestException('Asset not buyable via Instant');
+    } else if (('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.BANK) || !('paymentMethod' in dto)) {
+      if ('currency' in dto && !dto.currency.sellable) throw new BadRequestException('Currency not sellable via Bank');
```
Comment (davidleomay) [inline src/shared/services/payment-info.service.ts:38]: Just use else, without condition?

Comment (Yannick1712) [reply src/shared/services/payment-info.service.ts:38]: Haha sometimes it is so simple 😅 👍

**Review by davidleomay — APPROVED** [review]

**Pull request merged**